realm/realm-library/src/main/java/io/realm/internal/OsRealmConfig.java

```diff
@@ -222,6 +222,7 @@ private OsRealmConfig(final RealmConfiguration config,
         Byte clientResyncMode = (Byte) syncConfigurationOptions[j++];
         String encodedPartitionValue = (String) syncConfigurationOptions[j++];
         Object syncService = syncConfigurationOptions[j++];
+        Long appPtr = (Long) syncConfigurationOptions[j++];
```

Comment:
Not important, but would maybe make sense to keep the ordering in sync with the constructor arguments.